File: olo-promotions-sdk-csharp/Olo.Promotions.SDK/Requests/RedeemPromotionsRequest.cs
using System;
using System.Collections.Generic;
using Olo.Promotions.SDK.Requests.Models;
using Olo.Promotions.SDK.Requests.Models.Basket;
using Olo.Promotions.SDK.Requests.Models.Payments;

namespace Olo.Promotions.SDK.Requests
{
    public class RedeemPromotionsRequest
    {
        /// <summary>
        /// The ID of the order, which is only populated once the order has been placed.
        /// </summary>
        public string OrderId { get; set; }
        /// <summary>
        /// The loyalty account's unique identifier in the provider's system. Will only be provided in the case the basket or
        /// order contains loyalty rewards.
        /// </summary>
        public string AccountId { get; set; }
        /// <summary>
        /// The source of the order.
        /// <list type="table">
        ///     <listheader>
        ///         <term>Source</term>
        ///         <description>Value</description>
        ///     </listheader>
        ///     <item>
        ///         <term>Web (Desktop)</term>
        ///         <description>Web</description>
        ///     </item>
        ///     <item>
        ///         <term>Web (Mobile)</term>
        ///         <description>MobileWeb</description>
        ///     </item>
        ///     <item>
        ///         <term>Serve</term>
        ///         <description>MobileWeb</description>
        ///     </item>
        ///     <item>
        ///         <term>iOS</term>
        ///         <description>iOS</description>
        ///     </item>
        ///     <item>
        ///         <term>Android</term>
        ///         <description>Android</description>
        ///     </item>
        ///     <item>
        ///         <term>Kiosk</term>
        ///         <description>Kiosk</description>
        ///     </item>
        ///     <item>
        ///         <term>Not Listed</term>
        ///         <description>Other</description>
        ///     </item>
        /// </list>
        /// </summary>
        public Source Source { get; set; }
        /// <summary>
        /// The handoff method for the order.
        /// </summary>
        public Handoff? Handoff { get; set; }
        /// <summary>
        /// A three-letter <a href="https://www.iso.org/iso-4217-currency-codes.html">ISO 4217</a> code. If none is provided, will default to "USD".
        /// </summary>
        public string Currency { get; set; }
        /// <summary>
        /// A UTC date-time as defined in <a href="https://www.rfc-editor.org/rfc/rfc3339">RFC 3339</a> that represents when the order was created, but does not indicate when the order was paid for or when the guest wants to receive their food.
        /// </summary>
        public DateTime Placed { get; set; }
        /// <summary>
        /// A UTC date-time as defined in <a href="https://www.rfc-editor.org/rfc/rfc3339">RFC 3339</a> that represents when the guest wants to receive their food.
        /// </summary>
        public DateTime? Wanted { get; set; }
        /// <summary>
        /// The ID of the store as provided by the restaurant.
        /// </summary>
        public string StoreNumber { get; set; }
        /// <summary>
        /// A unique ID representing the vendor restaurant in Olo's system.
        /// </summary>
        public string Restaurant { get; set; }
        /// <summary>
        /// A unique ID representing the restaurant brand in Olo's system.
        /// </summary>
        public string Brand { get; set; }
        /// <summary>
        /// The cost of the food before applying any tax, tip, fees or discounts.
        /// </summary>
        public decimal Subtotal { get; set; }
        /// <summary>
        /// The amount of tax applied to the order.
        /// </summary>
        public decimal? Tax { get; set; }
        /// <summary>
        /// The amount the customer tipped on the order.
        /// </summary>
        public decimal? Tip { get; set; }
        /// <summary>
        /// The delivery fee for the order.
        /// </summary>
        public decimal? Delivery { get; set; }
        /// <summary>
        /// The sum of any custom fees applied to the order.
        /// </summary>
        public decimal? CustomFees { get; set; }
        /// <summary>
        /// The sum of any discounts applied to the order.
        /// </summary>
        public decimal? Discount { get; set; }
        /// <summary>
        /// The final total for the order after any tax, tip, fees and discounts.
        /// </summary>
        public decimal Total { get; set; }
        /// <summary>
        /// The destination address of the order.
        /// </summary>
        public Address Address { get; set; }
        /// <summary>
        /// Any payments applied to the order.
        /// </summary>
        public IEnumerable<Payment> Payments { get; set; } = new List<Payment>();
        /// <summary>
        /// The basket contents.
        /// </summary>
        public Basket Basket { get; set; }
    }
}

File: olo-promotions-sdk-go/pkg/promotions/models/requests/promotions/redeem_promotions_request.go
package promotions

import (
	"olo-promotions-sdk-go/pkg/promotions/models/requests"
	"time"
)

// RedeemPromotionsRequest represents a request to redeem promotions for an order.
type RedeemPromotionsRequest struct {
	// OrderID is the ID of the order, which is only populated once the order has been placed.
	OrderID string `json:"orderId"`

	// AccountID is the loyalty account's unique identifier in the provider's system.
	// Will only be provided if the basket or order contains loyalty rewards.
	AccountID string `json:"accountId"`

	// Source is the source of the order (e.g., Web, MobileWeb, iOS, etc.).
	Source requests.Source `json:"source"`

	// Handoff is the handoff method for the order (e.g., Pickup, Delivery, etc.).
	Handoff *requests.Handoff `json:"handoff,omitempty"`

	// Currency is a three-letter ISO 4217 code. Defaults to "USD" if not provided.
	Currency string `json:"currency"`

	// Placed is the UTC date-time when the order was created.
	Placed time.Time `json:"placed"`

	// Wanted is the UTC date-time when the guest wants to receive their food.
	Wanted *time.Time `json:"wanted,omitempty"`

	// StoreNumber is the ID of the store as provided by the restaurant.
	StoreNumber string `json:"storeNumber"`

	// Restaurant is a unique ID representing the vendor restaurant in Olo's system.
	Restaurant string `json:"restaurant"`

	// Brand is a unique ID representing the restaurant brand in Olo's system.
	Brand string `json:"brand"`

	// Subtotal is the cost of the food before applying any tax, tip, fees, or discounts.
	Subtotal float64 `json:"subtotal"`

	// Tax is the amount of tax applied to the order.
	Tax *float64 `json:"tax,omitempty"`

	// Tip is the amount the customer tipped on the order.
	Tip *float64 `json:"tip,omitempty"`

	// Delivery is the delivery fee for the order.
	Delivery *float64 `json:"delivery,omitempty"`

	// CustomFees is the sum of any custom fees applied to the order.
	CustomFees *float64 `json:"customFees,omitempty"`

	// Discount is the sum of any discounts applied to the order.
	Discount *float64 `json:"discount,omitempty"`

	// Total is the final total for the order after any tax, tip, fees, and discounts.
	Total float64 `json:"total"`

	// Address is the destination address of the order.
	Address *requests.Address `json:"address,omitempty"`

	// Payments are any payments applied to the order.
	Payments []requests.Payment `json:"payments"`

	// Basket is the basket contents of the order.
	Basket requests.Basket `json:"basket"`
}
